File: javascript/test/ExpressionFactoryTest.ts
import * as assert from 'node:assert'

import CucumberExpression from '../src/CucumberExpression'
import ExpressionFactory from '../src/ExpressionFactory'
import ParameterTypeRegistry from '../src/ParameterTypeRegistry'
import RegularExpression from '../src/RegularExpression'

describe('ExpressionFactory', () => {
  let expressionFactory: ExpressionFactory
  beforeEach(() => {
    expressionFactory = new ExpressionFactory(new ParameterTypeRegistry())
  })

  it('creates a RegularExpression', () => {
    assert.strictEqual(expressionFactory.createExpression(/x/).constructor, RegularExpression)
  })

  it('creates a CucumberExpression', () => {
    assert.strictEqual(expressionFactory.createExpression('x').constructor, CucumberExpression)
  })
})

File: dart/test/expression_factory_test.dart
import 'package:cucumber_expressions/src/cucumber_expression.dart';
import 'package:cucumber_expressions/src/expression_factory.dart';
import 'package:cucumber_expressions/src/parameter_type_registry.dart';
import 'package:cucumber_expressions/src/regular_expression.dart';
import 'package:test/test.dart';

void main() {
  group('ExpressionFactory', () {
    late ExpressionFactory expressionFactory;

    setUp(() {
      expressionFactory = ExpressionFactory(ParameterTypeRegistry());
    });

    test('creates a RegularExpression', () {
      expect(
        expressionFactory.createExpression(RegExp('x')),
        isA<RegularExpression>(),
      );
    });

    test('creates a CucumberExpression', () {
      expect(
        expressionFactory.createExpression('x'),
        isA<CucumberExpression>(),
      );
    });
  });
}
